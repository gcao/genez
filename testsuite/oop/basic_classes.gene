#!/usr/bin/env gene run

# Test: Basic class definition and usage
(class Point
  (.prop x)
  (.prop y))

# Create instance
(var p (new Point))
(= p/x 10)
(= p/y 20)

(print "Point x: " p/x)
(print "Point y: " p/y)

# Method definition
(fn Point_distance [self]
  (+ (* self/x self/x) (* self/y self/y)))

(print "Distance squared: " (p .distance))