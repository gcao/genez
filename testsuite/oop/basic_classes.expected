Point x:  10
Point y:  20
Distance squared:  500